astropy/modeling/core.py

```diff
@@ -255,6 +255,8 @@ def fset(self, value):
                     "instance or `None` (where `None` restores the default "
                     "inverse for this model if one is defined.")
 
+            if not isinstance(value, (_CompoundModel, type(None))):
```

Comment:
It's not super obvious to me a copy is made when assigning a custom inverse to a compound model:
```python
>>> gg = Gaussian1D(1, 0, 0.1) + Gaussian1D(1, 1, 0.2)
>>> p = Polynomial1D(4, c0=0.1, c1=0.2, c2=0.3, c3=0.4, c4=0.5)
>>> gg.inverse = p
>>> gg.inverse is p
True
```

So I think this check is unnecessary (it could just be `if value is not None`, say).